### PR TITLE
do not show runnables for main function outside of a binary target 

### LINKED ISSUE #4356: VSCode gives "run" and "debug" options for a "main" method that's not in a binary
I've got my application set up so that there is a `main.rs` with a `fn main()` to run the application, but this almost immediately delegates to an `fn main()` inside `lib.rs`. This lets me write integration tests against the library crate, but still build an executable crate at the end.

This works great, with only one tiny detail. 

rust-analyzer in VSCode is giving me the "run" and "debug" actions for the main function inside `lib.rs` as well as inside `main.rs`, but obviously you can't actually do those.

image

Clicking on this "run" action then gives me:
```
> Executing task in folder service: cargo run --package xxx --lib -- <

error: Found argument '--lib' which wasn't expected, or isn't valid in this context

USAGE:
    cargo run --package <SPEC>

For more information try --help
The terminal process terminated with exit code: 1
```

### PR DESCRIPTION
**Pull request by bnjjj**
close #4356

### CODE REVIEW

File: crates/rust-analyzer/src/main_loop/handlers.rs [hunk 1]
```diff
@@ -831,13 +843,23 @@ pub fn handle_code_lens(
...
             RunnableKind::Bench { .. } => "Run Bench",
-            RunnableKind::Bin => "Run",
+            RunnableKind::Bin => {
+                // Do not suggest binary run on other target than binary
+                match &cargo_spec {
+                    Some(spec) => match spec.target_kind {
+                        TargetKind::Bin => "Run",
+                        _ => continue,
+                    },
+                    None => continue,
```
Comment (bjorn3) [inline crates/rust-analyzer/src/main_loop/handlers.rs:860]: `cargo_spec` is None, when using `rust-analyzer.json`  instead of `cargo metadata`.

Comment (bnjjj) [reply crates/rust-analyzer/src/main_loop/handlers.rs:860]: Ok so I should also return "Run" in that case ?

Comment (bjorn3) [reply crates/rust-analyzer/src/main_loop/handlers.rs:860]: Yes

Comment (matklad) [reply crates/rust-analyzer/src/main_loop/handlers.rs:860]: tbh, for `rust-projec.json` I am not sure if runnables are at all useful. SO, for me the behavior there doesn't really matter

**Pull request merged**